src/test/java/be/seeseemelk/mockbukkit/entity/PlayerMockTest.java

```diff
@@ -928,4 +928,30 @@ public void testMultiplePotionEffects()
 			assertTrue(player.hasPotionEffect(effect.getType()));
 		}
 	}
+
+	@Test
+	public void testFirstPlayed() throws InterruptedException
+	{
+		PlayerMock player = new PlayerMock(server, "FirstPlayed123");
+
+		assertFalse(player.hasPlayedBefore());
+		assertEquals(0, player.getFirstPlayed());
+		assertEquals(0, player.getLastPlayed());
+		
+		server.addPlayer(player);
+		long firstPlayed = player.getFirstPlayed();
+		
+		assertTrue(player.hasPlayedBefore());
+		assertTrue(firstPlayed > 0);
+		assertEquals(player.getFirstPlayed(), player.getLastPlayed());
+		
+		// This forces the time to be different by at least 1ms
+		Thread.sleep(1);
```

Comment:
I'm personally not a big fan of something like this in a unit test, but I don't see any immediate way to prevent this.

Reply:
Yeah I am not a great fan of this either,
But I couldn't come up with any more elegant solution for this at the time, this was all done in a hurry.

But now after my first coffee, I may have come up with a better solution.